File: 2nd Year/Programming/Assignment 6 - CSharp Book Shop/Assignment 6 - CSharp Book Shop/Book.cs
using System;

namespace Bookshop
{
    public class Book : IComparable<Book>
    {
        public string ISBN { get; set; }
        public string Author { get; set; }
        public string Title { get; set; }
        public double Price { get; set; }

        public int CompareTo(Book obj)
        {
            /*
            return string.Compare(this.ISBN, obj.ISBN);
            */
            return string.Compare(this.Author, obj.Author);
            
        }
    }
}

File: 2nd Year/Programming/Assignment 2 - Book Shop/Assignment 2 - Book Shop/Book.cpp
#include "stdafx.h"
#include "Book.h"

CBook::CBook()
{

}

CBook::~CBook()
{

}

void CBook::setISBN(string isbn)
{
	ISBN = isbn;
}

void CBook::setTitle(string title)
{
	Title = title;
}

void CBook::setAuthor(string author)
{
	Author = author;
}

void CBook::setPrice(float price)
{
	Price = price;
}

string CBook::getISBN() const
{
	return ISBN;
}

string CBook::getTitle() const
{
	return Title;
}

string CBook::getAuthor() const
{
	return Author;
}

float CBook::getPrice() const
{
	return Price;
}

void CBook::getUserInput()
{
	cout << "Enter ISBN: ";
	cin >> ISBN;

	cout << "Enter Title: ";
	cin >> Title;

	cout << "Enter Price: ";
	cin >> Price;

	cout << "Enter Author: ";
	cin >> Author;
}

void CBook::displayBooks()
{
	cout << "\nISBN: " << ISBN << endl;
	cout << "\nTitle: " << Title << endl;
	cout << "\nAuthor: " << Author << endl;
	cout << "\nPrice: " << Price << endl;
}

